### ISSUE TITLE
Code-splitting guide missing commonly sought after answers.

### ISSUE THREAD

**Issue by rafde** [webpack/webpack.js.org#1366]
I was looking for the vendor common chunk example to reference and noticed that a lot of useful information was removed in https://github.com/webpack/webpack.js.org/commit/15f815589e863c7f7cd8d0c771cb83cb2e9ceec6#diff-97b9e4722b09b5046bf7756c344112cdL19
I can't find any examples or reference to where it was moved. 
Can that commit be reverted, as it offers nothing that people are actually looking for when visited.

**Comment by rafde** [comment]
atleast reference https://webpack.js.org/plugins/commons-chunk-plugin/#passing-the-minchunks-property-a-function in a prominent title that leads to that example.

**Comment by rouzbeh84** [comment]
@rafde I've gone ahead and noted that this is a commonly used feature of code splitting and reference the [Explicit vendor chunk](/plugins/commons-chunk-plugin/#explicit-vendor-chunk) explaining how to do this.

@skipjack I noticed a next steps section mentioning lazy-loading and caching. should there be a more robust optimization guide if there isn't one?

**Comment by skipjack** [comment]
I understand your frustration and I see that you were one of the authors of `code-splitting-libraries`. First off, I did credit all the previous authors in the new guide. Second of all, I had meant to ping those authors for review before merging #1338 but forgot -- so for that I apologize. Note that there was a 122-comment-long conversation on that PR prior to merging so you shouldn't think that we took the decision to make those changes lightly.

Here are a few reasons why the changes were made:

- There were no basic examples on the various approaches to code splitting itself (#1333).
- Much, if not all, of the documentation discussed in the previous articles was in relation to caching -- for which there is already a separate guide (that still needs to be reviewed and re-worked -- #1258 and #652). 
- A lot of the documentation from `code-splitting-async` was really detailed reference documentation and didn't make sense in a guide.
- All three of the previous guides were completely out of sync with the examples we're trying to build on from getting-started onwards (#1258).

> Can that commit be reverted, as it offers nothing that people are actually looking for when visited.
> webpack docs are being neutered...

I think if you really think about it and see my comments above, you'll realize these comments are a bit overboard. I'll review @rouzbeh84's PR now to see if we can get a short-term fix in and note it in #1258 so we don't forget to address it. Again, as I said on gitter, please be a little more thoughtful in your approach. Just because your use case is splitting out vendor code, doesn't mean code splitting can't be used for a variety of other things. Anything missing from the caching guide now should be filled in soon -- if you want, I can even ping you for a review once it's in progress.

As for "the plan" you mentioned on gitter -- please see #1380. This is something we've discussed offline for awhile now but didn't have time to add until tonight.

**Comment by rafde** [comment]
> please be a little more thoughtful in your approach
Bringing in the gitter discussions has no context. That's why I made this issue, so I am going to focus why I made this issue.

One of the things I noticed people complain about webpack is about documentation. 

So when I saw the changes I thought to myself: There goes one of the most looked at [webpack question on stackoverflow](https://stackoverflow.com/questions/tagged/webpack?sort=votes&pageSize=15#question-summary-30329337). 
When people ask webpack questions, I want to direct them to the doc site.  
I looked over all you links and https://github.com/webpack/webpack.js.org/issues/1333 make sense as an issue, but I don't know if that overall issue really dwarfs what the majority of the people focus on finding relative to that stack overflow question I referenced. Maybe analytics can shed some light.

Anyhow, do what you gotta do. I stopped directing people to the doc site and use @bebraw 's guide for anything extra that I want to start getting into regarding webpack. He does a better job with guides since it's his :bread: and butter.

:peace_symbol: and :four_leaf_clover:
